src/ducks/account.ts

```diff
@@ -33,58 +38,137 @@ export const fetchAccountAction = createAsyncThunk<
         } as UnfundedAccount;
+        isUnfunded = true;
       }
     } catch (error) {
+      log.error({
+        title: `Fetching account ${publicKey} failure`,
+        body: error,
+      });
+
       return rejectWithValue({
         errorString: getErrorString(error),
       });
     }
 
-    return { data: stellarAccount, secretKey };
+    log.response({
+      title: `Account info fetched`,
+      body: stellarAccount,
+    });
+
+    return { data: stellarAccount, assets, isUnfunded, secretKey };
   },
 );
 
-export const createRandomAccountAndFundIt = createAsyncThunk<
+export const createRandomAccount = createAsyncThunk<
   string,
   undefined,
   { rejectValue: RejectMessage; state: RootState }
->("account/createRandomAccountAndFundIt", async (_, { rejectWithValue }) => {
+>("account/createRandomAccount", (_, { rejectWithValue }) => {
   try {
+    log.instruction({ title: "Generating new keypair" });
     const keypair = Keypair.random();
-    await fetch(`https://friendbot.stellar.org?addr=${keypair.publicKey()}`);
-
     return keypair.secret();
   } catch (error) {
+    log.error({
+      title: "Generating new keypair failed",
+      body: error.toString(),
+    });
     return rejectWithValue({
-      errorString: "Something went wrong, please try again.",
+      errorString:
+        "Something went wrong while creating random account, please try again.",
     });
   }
 });
 
+export const fundTestnetAccount = createAsyncThunk<
+  { data: Types.AccountDetails; assets: AnyObject; isUnfunded: boolean },
+  string,
+  { rejectValue: RejectMessage; state: RootState }
+>(
+  "account/fundTestnetAccount",
+  async (publicKey, { rejectWithValue, getState }) => {
+    log.instruction({
+      title: "The friendbot is starting to fund your testnet account",
+    });
+
+    const { pubnet } = settingsSelector(getState());
+    const networkConfig = getNetworkConfig(pubnet);
+
+    const dataProvider = new DataProvider({
+      serverUrl: networkConfig.url,
+      accountOrKey: publicKey,
+      networkPassphrase: networkConfig.network,
+    });
+
+    try {
+      await fetch(`https://friendbot.stellar.org?addr=${publicKey}`);
+      const stellarAccount = await dataProvider.fetchAccountDetails();
+      const assets = await getAssetData({
+        balances: stellarAccount.balances,
+        networkUrl: networkConfig.url,
+      });
+
+      log.response({
+        title: "The friendbot funded your account",
+        body: stellarAccount,
+      });
+
+      return { data: stellarAccount, assets, isUnfunded: false };
+    } catch (error) {
+      log.error({
+        title: "The friendbot funding failed",
+        body: error.toString(),
+      });
+
+      return rejectWithValue({
+        errorString:
+          "Something went wrong funding the account, please try again.",
+      });
+    }
+  },
+);
+
 const initialState: AccountInitialState = {
   data: null,
+  assets: {},
   errorString: undefined,
   isAuthenticated: false,
```

Comment:
This is for UI only. I think it's more convenient and easier to remember to use `account.isAuthenticated` than manually check for any other prop. Especially if the app grows and there are more places where this would be used.